packages/react-instantsearch-dom-maps/src/Control.js

```diff
@@ -1,65 +1,58 @@
-import { STATE_CONTEXT } from './Provider';
+import GeoSearchContext from './GeoSearchContext';
 import withGoogleMaps from './withGoogleMaps';
 
 const cx = createClassNames('GeoSearch');
+const ControlPropTypes = {
+  googleMapsInstance: PropTypes.object.isRequired,
+  translate: PropTypes.func.isRequired,
+};
 
-export class Control extends Component {
-  static propTypes = {
-    googleMapsInstance: PropTypes.object.isRequired,
-    translate: PropTypes.func.isRequired,
-  };
+export const Control = ({ googleMapsInstance, translate, context }) => (
+  <div className={cx('control')}>
+    {context.isRefineOnMapMove || !context.hasMapMoveSinceLastRefine ? (
+      <label className={cx('label')}>
+        <input
+          className={cx('input')}
+          type="checkbox"
+          checked={context.isRefineOnMapMove}
+          onChange={context.toggleRefineOnMapMove}
+        />
+        {translate('control')}
+      </label>
+    ) : (
+      <button
+        className={cx('redo')}
+        onClick={() => context.refineWithInstance(googleMapsInstance)}
+      >
+        {translate('redo')}
+      </button>
+    )}
+  </div>
+);
 
-  static contextTypes = {
-    [STATE_CONTEXT]: PropTypes.shape({
-      isRefineOnMapMove: PropTypes.bool.isRequired,
-      toggleRefineOnMapMove: PropTypes.func.isRequired,
-      hasMapMoveSinceLastRefine: PropTypes.bool.isRequired,
-      refineWithInstance: PropTypes.func.isRequired,
-    }).isRequired,
-  };
+Control.propTypes = {
+  ...ControlPropTypes,
+  context: PropTypes.shape({
+    isRefineOnMapMove: PropTypes.bool.isRequired,
+    toggleRefineOnMapMove: PropTypes.func.isRequired,
+    hasMapMoveSinceLastRefine: PropTypes.bool.isRequired,
+    refineWithInstance: PropTypes.func.isRequired,
+  }),
+};
 
-  getStateContext() {
-    return this.context[STATE_CONTEXT];
-  }
+const ControlWrapper = props => (
+  <GeoSearchContext.Consumer>
+    {context => <Control {...props} context={context} />}
+  </GeoSearchContext.Consumer>
+);
 
-  render() {
-    const { googleMapsInstance, translate } = this.props;
-    const {
-      isRefineOnMapMove,
-      hasMapMoveSinceLastRefine,
-      toggleRefineOnMapMove,
-      refineWithInstance,
-    } = this.getStateContext();
-
-    return (
-      <div className={cx('control')}>
-        {isRefineOnMapMove || !hasMapMoveSinceLastRefine ? (
-          <label className={cx('label')}>
-            <input
-              className={cx('input')}
-              type="checkbox"
-              checked={isRefineOnMapMove}
-              onChange={toggleRefineOnMapMove}
-            />
-            {translate('control')}
-          </label>
-        ) : (
-          <button
-            className={cx('redo')}
-            onClick={() => refineWithInstance(googleMapsInstance)}
-          >
-            {translate('redo')}
-          </button>
-        )}
-      </div>
-    );
-  }
-}
+ControlWrapper.displayName = 'Control';
```

Comment:
I don't think it's valuable to rename it, we have two times the same component in the tree.

![screenshot 2019-02-14 at 10 10 48](https://user-images.githubusercontent.com/6513513/52775836-e0426a80-3040-11e9-9e9c-c006dc04c62d.png)